pkg/autodiscovery/listeners/workloadmeta.go

```diff
@@ -93,6 +95,7 @@ func (l *workloadmetaListenerImpl) Store() workloadmeta.Store {
 }
 
 func (l *workloadmetaListenerImpl) AddService(svcID string, svc Service, parentSvcID string) {
+	kind := kindFromSvcID(svcID)
```

Comment:
Might be useful to change the func signature to receive a `service` (private struct used by all workloadmeta listeners) instead of `Service` (the generic public interface), and use its `entity.GetID().Kind` to prevent future breakage.

Reply:
You're referring to changing `AddService`'s signature if I understood correctly?

Reply:
It's a good idea, it's just that it would require caching the children kinds so we can know their kinds when removed during an unset event of their parent object. Not sure it's worth the extra overhead 🤔

Reply:
Ah right, turns out to be a bit more work than I expected. Nevermind for now :shipit: